src/CommunityToolkit.Maui/Views/AvatarView.shared.cs

```diff
@@ -354,12 +354,32 @@ void HandlePropertyChanged(object? sender, PropertyChangedEventArgs e)
 			&& Width >= 0 // The default value of Y (before the view is drawn onto the page) is -1
 			&& avatarImage.Source is not null)
 		{
-			var imageWidth = Width - (StrokeThickness * 2) - Padding.Left - Padding.Right;
-			var imageHeight = Height - (StrokeThickness * 2) - Padding.Top - Padding.Bottom;
+			double imageWidth = Width - (StrokeThickness * 2) - Padding.Left - Padding.Right;
+			double imageHeight = Height - (StrokeThickness * 2) - Padding.Top - Padding.Bottom;
 
 			var rect = new Rect(0, 0, imageWidth, imageHeight);
 
-			avatarImage.Clip = new RoundRectangleGeometry { CornerRadius = CornerRadius, Rect = rect };
+			switch (StrokeShape)
+			{
+				case Polyline polyLine:
+					avatarImage.Clip = polyLine.Clip;
```

Comment:
I think you can use switch expression

Reply:
I did have it, but went with the switch whilst I has working thru the problem. I'll try and remember to convert when done.